articlequality/feature_lists/ptwiki.py

```diff
@@ -0,0 +1,169 @@
+"""
+Portuguese Wikipedia
+++++++++++++++++++++
+"""
+import re
+
+from revscoring import Feature
+from revscoring.datasources.meta import filters, mappers
+from revscoring.features import wikitext
+from revscoring.features.meta import aggregators
+from revscoring.features.modifiers import log, max, sub
+from revscoring.languages import portuguese
+
+from . import wikipedia
+
+# Templates
+infobox_templates = wikitext.revision.template_names_matching(
+    r"Info", name="ptwiki.revision.infobox_templates")
+CN_TEMPLATES = [
+    r"Carece de fontes",
+    r"Carece de fontes2",
+    r"Carece de fontes/bloco",
+    r"Carece de fontes/bloco2"
+]
+cn_templates = wikitext.revision.template_names_matching(
+    "|".join(CN_TEMPLATES), name="ptwiki.revision.cn_templates")
+MAIN_TEMPLATES = [
+    r"Artigo[ _]principal",
+    r"Ver[ _]artigo[ _]principali",
+    r"Principal",
+    r"Ver[ _]também artigo[ _]principal",
+    r"Main",
+    r"Detalhes",
+    r"Mais",
+    r"Artigoprincipal",
+    r"AP", r"Details", r"Ver[ _]artigo"
+]
+main_article_templates = wikitext.revision.template_names_matching(
+    "|".join(MAIN_TEMPLATES), name="ptwiki.main_article_templates")
+CITE_TEMPLATES = [
+    r"Cite",
+    r"Citar",
+    r"Citar web",
+    r"Citar livro",
+    r"Harvard[_ ]citation[_ ]no[_ ]brackets", r"harvnb",
+    r"Harvard citation", r"harv",
+    r"Harvard citation text", r"harvtxt",
+    r"Harvcoltxt",
+    r"Harvcol",
+    r"Harvcolnb",
+    r"Harvard citations", r"harvs",
+    r"Harvp"
+]
+cite_templates = wikitext.revision.template_names_matching(
+    "|".join(CITE_TEMPLATES), name="ptwiki.revision.cite_templates")
+shortened_footnote_templates = wikitext.revision.template_names_matching(
+    r"sfn", name="ptwiki.revision.shortened_footnote_templates")
+all_ref_tags = shortened_footnote_templates + wikitext.revision.ref_tags
+all_cite_templates = cite_templates + shortened_footnote_templates
+proportion_of_templated_references = \
+    all_cite_templates / max(all_ref_tags, 1)
+non_templated_references = max(all_ref_tags - all_cite_templates, 0)
+non_cite_templates = sub(
+    wikitext.revision.templates, all_cite_templates,
+    name="ptwiki.revision.non_cite_templates"
+)
+
+# Links
+category_links = wikitext.revision.wikilink_titles_matching(
+    r"(Category|Categoria)\:", name="ptwiki.revision.category_links")
```

Comment:
Considering that MediaWiki also accepts `[[Category             :Example]]`, maybe add `\s*` to match spaces before `\:`? (I assume `mwparserfromhell` does not normalize the titles before comparing to the regex)